### PR TITLE
DM-7325: Port to Python 3

### PR DESCRIPTION
**Pull request by fred3m**

### CODE REVIEW

File: tests/testEMPsfApprox.py [hunk 1]
```diff
@@ -174,24 +182,32 @@ def testEMPlugin(self):
...
         #  some kind of reasonable value for the fit.  A more quantitative test may be needed.
-        self.assertClose(flux0/flux1, 7.0/3.0, rtol = .05)
-        self.assertClose(comp0.getEllipse().getCore().getIxx(), 16.0, rtol = .05)
-        self.assertClose(comp0.getEllipse().getCore().getIyy(), 16.0, rtol = .05)
-        self.assertClose(comp0.getEllipse().getCore().getIxy(), 0.0, atol = .1)
-        self.assertClose(comp1.getEllipse().getCore().getIxx(), 100.0, rtol = .05)
-        self.assertClose(comp1.getEllipse().getCore().getIyy(), 100.0, rtol = .05)
-        self.assertClose(comp1.getEllipse().getCore().getIxy(), 0.0, atol = .1)
+        self.assertClose(flux0/flux1, 7.0/3.0, rtol=.05)
+        self.assertClose(comp0.getEllipse().getCore().getIxx(), 16.0, rtol=.05)
```
Comment (timj) [inline tests/testEMPsfApprox.py]: `assertFloatsAlmostEqual`

File: python/lsst/meas/extensions/ngmix/LMSimpleShape.py [hunk 2]
```diff
@@ -150,7 +151,8 @@ def measure(self, measRecord, exposure):
             shape = psfArray.shape
             image = lsst.afw.image.ImageD(shape[1], shape[0])
             evaluate = measRecord.get(self.psfMsfKey).evaluate()
-            evaluate.addToImage(image.getArray(), lsst.afw.geom.Point2I(-(shape[0] - 1)//2, -(shape[1] - 1)//2))
+            evaluate.addToImage(
+                image.getArray(), lsst.afw.geom.Point2I(-(shape[0] - 1)//2, -(shape[1] - 1)//2))
```
Comment (timj) [inline python/lsst/meas/extensions/ngmix/LMSimpleShape.py]: this change looks weird. Move the first argument up a line and just have the second argument on the new line with proper indenting.

File: python/lsst/meas/extensions/ngmix/LMSimpleShape.py [hunk 3]
```diff
@@ -206,19 +207,23 @@ def measure(self, measRecord, exposure):
                     SingleFrameLMSimpleShapePlugin.ErrEnum.flag_singularMatrix)
             if res['flags'] & LM_NEG_COV_EIG:
                 raise MeasurementError(
-                    self.flagHandler.getDefinition(SingleFrameLMSimpleShapePlugin.ErrEnum.flag_negativeCovEig).doc,
+                    self.flagHandler.getDefinition(
+                        SingleFrameLMSimpleShapePlugin.ErrEnum.flag_negativeCovEig).doc,
```
Comment (timj) [inline python/lsst/meas/extensions/ngmix/LMSimpleShape.py]: The commas and parens were confusing me a bit. Does this really need to go on a new line? How long was it without?

File: tests/testEMPsfApprox.py [hunk 4]
```diff
@@ -32,43 +33,49 @@
 import lsst.afw.table as afwTable
 import lsst.meas.algorithms as measAlg
 import lsst.meas.base as measBase
-import lsst.meas.base.tests
+from lsst.meas.base.tests import AlgorithmTestCase
 
 import lsst.meas.extensions.ngmix.EMPsfApprox
 
 #   Create an array of size x size containing a 2D circular Gaussian of size sigma.  Normalized to 1.0
```
Comment (timj) [inline tests/testEMPsfApprox.py]: put this comment inside the function as a docstring.

File: tests/testEMPsfApprox.py [hunk 5]
```diff
@@ -32,43 +33,49 @@
...
-        for xi, xv in enumerate(xrange(0, size)):
+    image = afwImage.ImageD(afwGeom.Box2I(afwGeom.Point2I(0, 0), afwGeom.Point2I(size, size)))
+    array = np.ndarray(shape=(size, size), dtype=np.float64)
+    for yi, yv in enumerate(range(0, size)):
+        for xi, xv in enumerate(range(0, size)):
             array[yi, xi] = np.exp(-0.5*((xv - xc)**2 + (yv - yc)**2)/sigma**2)
     array /= array.sum()
     return array
 
 #   Run a measurement task which has previously been initialized on a single source
```
Comment (timj) [inline tests/testEMPsfApprox.py]: comment to docstring.

File: tests/testEMPsfApprox.py [hunk 6]
```diff
@@ -32,43 +33,49 @@
...
     # doesn't appear on the console or in logs, and incorrectly cause the user
     # to assume a failure.
-    task.log.setThreshold(task.log.FATAL)
+    task.log.setLevel(task.log.FATAL)
     footprints = dettask.detectFootprints(exposure, sigma=4.0).positive.getFootprints()
     source.setFootprint(footprints[0])
     task.run(exposure, cat)
     return source
 
 #   make a Gaussian with one or two components.  Always square of dimensions size x size
```
Comment (timj) [inline tests/testEMPsfApprox.py]: comment to docstring (which lets you remove the two blank lines as well -- also previous).

File: python/lsst/meas/extensions/ngmix/LMSimpleShape.py (hunk 2 above)
Comment (fred3m) [reply python/lsst/meas/extensions/ngmix/LMSimpleShape.py]: ok

File: python/lsst/meas/extensions/ngmix/LMSimpleShape.py (hunk 3 above)
Comment (fred3m) [reply python/lsst/meas/extensions/ngmix/LMSimpleShape.py]: A single line is 135 characters. I could change it to 

``` python
raise MeasurementError(self.flagHandler.getDefinition(
                       SingleFrameLMSimpleShapePlugin.ErrEnum.flag_negativeCovEig).doc,
                       SingleFrameLMSimpleShapePlugin.ErrEnum.flag_negativeCovEig)
```

which looks a little better but could be more confusing. I don't mind changing it though.

Comment (timj) [reply python/lsst/meas/extensions/ngmix/LMSimpleShape.py]: Yes, I am torn. Whatever `flake8` allows is fine with me.

File: tests/testEMPsfApprox.py (hunk 4 above)
Comment (fred3m) [reply tests/testEMPsfApprox.py]: ok

File: tests/testEMPsfApprox.py (hunk 5 above)
Comment (fred3m) [reply tests/testEMPsfApprox.py]: ok

File: tests/testEMPsfApprox.py (hunk 6 above)
Comment (fred3m) [reply tests/testEMPsfApprox.py]: ok

**Pull request merged**